osbenchmark/workload_generator/extractors.py

```diff
@@ -0,0 +1,223 @@
+        self.client = client
+
+    def template_vars(self,index_name, out_path, doc_count):
+        comp_outpath = out_path + COMP_EXT
+        return {
+            "index_name": index_name,
+            "filename": os.path.basename(comp_outpath),
+            "path": comp_outpath,
+            "doc_count": doc_count,
+            "uncompressed_bytes": os.path.getsize(out_path),
+            "compressed_bytes": os.path.getsize(comp_outpath)
+        }
+
+    def _get_doc_outpath(self, outdir, name, suffix=""):
+        return os.path.join(outdir, f"{name}-documents{suffix}.json")
+
+
+    def extract_documents(self, index, documents_limit=None):
+        """
+        Scroll an index with a match-all query, dumping document source to ``outdir/documents.json``.
+
+        :param index: Name of index to dump
+        :param documents_limit: The number of documents to extract. Must be equal
+        to or less than the total number of documents that exists in the index
+        :return: dict of properties describing the corpus for templates
+        """
+
+        logger = logging.getLogger(__name__)
+
+        total_documents = self.client.count(index=index)["count"]
+
+        documents_to_extract = total_documents if not documents_limit else min(total_documents, documents_limit)
+
+        if documents_to_extract > 0:
+            logger.info("[%d] total docs in index [%s]. Extracting [%s] docs.", total_documents, index, documents_to_extract)
+            docs_path = self._get_doc_outpath(self.custom_workload.workload_path, index)
+            # Create test mode corpora
+            self.dump_documents(
+                self.client,
+                index,
+                self._get_doc_outpath(self.custom_workload.workload_path, index, "-1k"),
+                min(documents_to_extract, 1000),
+                " for test mode")
+            # Create full corpora
+            self.dump_documents(self.client, index, docs_path, documents_to_extract)
+
+            return self.template_vars(index, docs_path, documents_to_extract)
+        else:
+            logger.info("Skipping corpus extraction fo index [%s] as it contains no documents.", index)
+            return None
+
+
+    def dump_documents(self, client, index, out_path, number_of_docs, progress_message_suffix=""):
+        # pylint: disable=import-outside-toplevel
+        from opensearchpy import helpers
+
+        logger = logging.getLogger(__name__)
+        freq = max(1, number_of_docs // 1000)
+
+        progress = console.progress()
+        compressor = DOCS_COMPRESSOR()
+        comp_outpath = out_path + COMP_EXT
+        with open(out_path, "wb") as outfile:
+            with open(comp_outpath, "wb") as comp_outfile:
+                logger.info("Dumping corpus for index [%s] to [%s].", index, out_path)
+                query = {"query": {"match_all": {}}}
+                for n, doc in enumerate(helpers.scan(client, query=query, index=index)):
```

Comment:
Nit: generally `i`, `j`, `k` are appropriate names for changing values, `n` is usually the target.